components/director/pkg/webhook_client/client.go

```diff
@@ -238,9 +239,9 @@ func parseResponseObject(resp *http.Response) (*webhook.ResponseObject, error) {
 	}, nil
 }
 
-func checkForErr(resp *http.Response, successStatusCode *int, errorMessage *string) error {
+func checkForErr(resp *http.Response, successStatusCode, incompleteStatusCode *int, errorMessage *string) error {
 	var errMsg string
-	if *successStatusCode != resp.StatusCode {
+	if *successStatusCode != resp.StatusCode && (incompleteStatusCode == nil || *incompleteStatusCode != resp.StatusCode) {
```

Comment:
Instead of checking for nil here, there is a Validate method of the response and do the nil check there. And here check only the status codes.